cmp/grid/columns/Column.js

```diff
@@ -377,14 +378,24 @@ export class Column {
         } else if (this.tooltipElement) {
             ret.tooltipValueGetter = ({value}) => value;
             ret.tooltipComponentFramework = class extends Component {
-                getReactContainerClasses() {return ['xh-grid-tooltip']}
+                getReactContainerClasses() {
+                    const agParams = this.props,
+                        {location} = agParams;
+                    return location === 'header' ?
+                        ['ag-tooltip'] :
+                        ['xh-grid-tooltip'];
+                }
                 render() {
                     const agParams = this.props,
-                        {value, rowIndex, api} = agParams,
-                        record = api.getDisplayedRowAtIndex(rowIndex).data;
+                        {rowIndex, api, colDef, location} = agParams;
+
+                    if (location === 'header') {
+                        return div(colDef.headerTooltip);
+                    }
 
-                    // ag-Grid encodes the value, so we decode it before passing to the renderer
-                    return me.tooltipElement(decodeURIComponent(value), {record, column: me, gridModel, agParams});
+                    const record = api.getDisplayedRowAtIndex(rowIndex).data,
```

Comment:
do we always get a valid rowIndex?  Do we always get a Record?  What about when the user mouses over a group row or other interstitial space?  I would feel more comfortable if this code was on the lookout for that, and rendered nothing rather than a DNP.

Reply:
I also think we should be using the value getter on the column.  Its called out in constructor, and while it defaults to what you have here, it can be overridden by users:

 * @param {Column~getValueFn} [c.getValueFn] - function for getting the column value